samplePDF/samplePDFFDBase.cpp

```diff
@@ -207,120 +332,55 @@ bool samplePDFFDBase::IsEventSelected(const std::vector< std::string >& Paramete
-void samplePDFFDBase::calcOscWeights(int sample, int nutype, double *w)
-{
-  MCSamples[sample].Oscillator->setMNSMatrix(asin(sqrt(*oscpars[0])),asin(sqrt(*oscpars[2])), asin(sqrt(*oscpars[1])), (*oscpars[5]), nutype);
-  MCSamples[sample].Oscillator->setNeutrinoMasses(*oscpars[3], *oscpars[4]);
-  MCSamples[sample].Oscillator->calculateProbabilities(MCSamples[sample].NeutrinoType);
-  MCSamples[sample].Oscillator->getProbabilityArr(w, MCSamples[sample].ProbType);
-}
-#endif 
-
 void samplePDFFDBase::reweight() // Reweight function - Depending on Osc Calculator this function uses different CalcOsc functions
 {
-
-  if (Osc!=NULL) {
-	std::cout << "Osc is not NULL!! i.e. doing atm oscillations " << std::endl;
-    //DB Currently hardcoded to assume rho_electrons = rho_matter/2, 25km production height
-    Osc->FillOscillogram(oscpars,25.0,0.5);
-    for (unsigned int iSample=0;iSample<MCSamples.size();iSample++) {
-      for (int iEvent=0;iEvent<MCSamples[iSample].nEvents;iEvent++) {
-		MCSamples[iSample].osc_w[iEvent] = *(MCSamples[iSample].osc_w_pointer[iEvent]);
-      }
-    }
-  } else {
-    for (int i=0; i< (int)MCSamples.size(); ++i) {
-      
-#if defined (USE_PROB3) && defined (CPU_ONLY)
-      //Prob3 CPU needs to loop through events too
-      for(int j = 0; j < MCSamples[i].nEvents; ++j) {
-		MCSamples[i].osc_w[j] = calcOscWeights(i, MCSamples[i].nutype, MCSamples[i].oscnutype, *(MCSamples[i].rw_etru[j]));
-      } //event loop
-#endif
-      
-#if defined (USE_PROB3) && not defined (CPU_ONLY)
-      calcOscWeights(MCSamples[i].nutype, MCSamples[i].oscnutype, *(MCSamples[i].rw_etru), MCSamples[i].osc_w, MCSamples[i].nEvents);
-#endif
-      
-#if not defined (USE_PROB3)
-      calcOscWeights(i, MCSamples[i].nutype, MCSamples[i].osc_w);
-#endif
-    }// Sample loop
-  }
-
   //KS: Reset the histograms before reweight 
   ResetHistograms();
-
+  
+  std::vector<_float_> OscVec(OscCov->GetNumParams());
+  for (int iPar=0;iPar<OscCov->GetNumParams();iPar++) {
+    OscVec[iPar] = OscCov->getParProp(iPar);
+  } 
+  for (int iSample=0;iSample<(int)MCSamples.size();iSample++) {
+    NuOscProbCalcers[iSample]->CalculateProbabilities(OscVec);
+  }
+  
   fillArray();
-
+  
   return;
 }
 
-// ************************************************
-//DB Function which does the core reweighting. This assumes that oscillation weights have already been calculated and stored in samplePDFFDBase[iSample].osc_w[iEvent]
-//This function takes advantage of most of the things called in setupSKMC to reduce reweighting time
-//It also follows the ND code reweighting pretty closely
-//This function fills the samplePDFFD_array array which is binned to match the sample binning, such that bin[1][1] is the equivalent of _hPDF2D->GetBinContent(2,2) {Noticing the offset}
+//************************************************
+/// @function samplePDFFDBase::fillArray()
+/// Function which does the core reweighting. This assumes that oscillation weights have 
```

Comment:
why is the function that does the core reweighting called `fillArray` and there is another function called `reweight`, which apparently doesn't do the core reweighting? or does some part of it?